cmd/kubeadm/app/phases/upgrade/staticpods.go

```diff
@@ -263,11 +264,19 @@ func performEtcdStaticPodUpgrade(client clientset.Interface, waiter apiclient.Wa
 	}
 
 	// gets the etcd version of the local/stacked etcd member running on the current machine
+	// the version is read from che cluster; this should take into account that there are still
+	// around old etcd manifest with etcd listening on local host only
+	// N.B. taking care of old etcd manifests is necessary only in v1.14; starting from v1.15 all the etcd manifest should have 2 endpoints
+	etcdURL := etcdutil.GetClientURL(&cfg.LocalAPIEndpoint)
+	if etcdutil.IsEtcdListeningOnLocalHostOnly() {
+		etcdURL = "localhost:2379"
```

Comment:
this could be `127.0.0.1:2379` as well right?

Reply:
`localhost:2379` is what I get if the etcd is listening on localhost only, even if the `--listen-client-urls=https://127.0.0.1:2379`. 

Now that I think about this, I think that when etcd is listening on localhost only it is safer to simply get the etcd version from the first element in `currentEtcdVersions` (that should be the only one) instead of checking for an element with a specific address.

@rosti @mauilion @neolit123 opinions?

Reply:
@fabriziopandini it will be OK because we don't support mismatched versions:
https://github.com/kubernetes/kubernetes/blob/9c75270b2be4dcc39e31edb34b76e8f604aaf91a/cmd/kubeadm/app/util/etcd/etcd.go#L262

Reply:
This got me thinking, I looked at etcd and etcd client code (not that hard I admit), but I couldn't find any use of `LookupAddr` or something, that would turn 127.0.0.1 from ClientURLs into localhost.
Yet it's working (I tested it yesterday).

Reply:
non blocking for merge.